websites/W/Wikimedia/presence.ts

```diff
@@ -0,0 +1,551 @@
+            "Iberocoop talk": "Viewing an Iberocoop talk page",
+            Tool: "Viewing a tool page",
+            "Tool talk": "Viewing a tool talk page",
+            Module: "Viewing a module page",
+            "Module talk": "Viewing a module talk page",
+            Translations: "Viewing a translations page",
+            "Translations talk": "Viewing a translations talk page",
+            Newsletter: "Viewing a newsletter page",
+            "Newsletter talk": "Viewing a newsletter talk page",
+            CNBanners: "Viewing a CNBanners page",
+            "CNBanners talk": "Viewing a CNBanners talk page",
+            Schema: "Viewing a schema page",
+            "Schema talk": "Viewing a schema talk page",
+            Research: "Viewing a research page",
+            "Research talk": "Viewing a research talk page",
+            Grants: "Viewing a grants page",
+            "Grants talk": "Viewing a grants talk page",
+            Manual: "Viewing a manual page",
+            "Manual talk": "Viewing a manual talk page",
+            Form: "Viewing a form page",
+            "Form talk": "Viewing a form talk page"
+          },
+            canonicalNamespace = mwConfig.wgCanonicalNamespace.replace(
+              /_/g,
+              " "
+            );
+          return (
+            details[mwConfig.wgNamespaceNumber] ||
+            details[canonicalNamespace] ||
+            `Viewing a/an ${canonicalNamespace} page`
+          );
+
+          /* Witheld until we found a better solution for future translating */
+
+          // const details: {[index: string]: string} = {
+          // 	"-2": "Viewing a media",
+          // 	0: "Reading an article",
+          // 	1: "Viewing a talk page",
+          // }
+
+          // const detailsShort: {[index: string]: string} = {
+          // 	"Special": "a special",
+          // 	"User": "a user",
+          // 	"Project": "a project",
+          // 	"File": "a file",
+          // 	"Interface": "an interface",
+          // 	"Template": "a template",
+          // 	"Help": "a help",
+          // 	"Category": "a category",
+          // 	"Portal": "a portal",
+          // 	"Draft": "a draft",
+          // 	"Module": "a module",
+          // 	"Gadget": "a gadget",
+          // 	"Gadget definition": "a gadget defintion",
+          // 	"Manual": "a manual",
+          // 	"Topic": "a topic"
+          // }
+
+          // const namespaceName = mwConfig.wgCanonicalNamespace.replace(/_/g, " ")
+          // return details[mwConfig.wgNamespaceNumber] || `Viewing ${detailsShort[namespaceName.split(" talk")[0]] + (namespaceName.endsWith(" talk") ? " talk" : "") || `a/n ${namespaceName}`} page`
+        };
+
+      //
+      // Important note:
+      //
+      // When checking for the current location, avoid using the URL.
+      // The URL is going to be different in other languages.
+      // Use the elements on the page instead.
+      //
```

Comment:
no comments

Reply:
Really weird rule you got there... Removed nonetheless.